src/Workspaces/Core/Portable/Workspace/WorkspaceEventOptions.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis;
+
+internal record struct WorkspaceEventOptions(bool RequiresMainThread)
+{
+    public static readonly WorkspaceEventOptions DefaultOptions = new(RequiresMainThread: false);
+    public static readonly WorkspaceEventOptions RequiresMainThreadOptions = new(RequiresMainThread: true);
```

Comment:
On one hand, call it the "main thread" is bugging me here since from the workspace perspective it doesn't really have a concept of the main thread like JTF, it's just "the scheduler we schedule things to". On the other hand, we all know what this means and I can't come up with a better name.

Reply:
Will keep it for now as I can't come up with something better (foreground, ui, main all have this same association).